src/dependabot/update_metadata.ts

```diff
@@ -49,7 +49,11 @@ export async function parse (commitMessage: string, body: string, branchName: st
 
     if (data['updated-dependencies']) {
       return await Promise.all(data['updated-dependencies'].map(async (dependency, index) => {
-        const dirname = `/${chunks.slice(2, -1 * (1 + (dependency['dependency-name'].match(/\//g) || []).length)).join(delim) || ''}`
+        // When a branch delimiter of "-" is used, we need to +1 to end of slice because there is always a hyphen
+        // between dependency name and version at the end of the branch name, regardless of configured branch separator.
+        // e.g. "fsevents-1.2.13".
+        const baseSliceEnd = delim === '-' ? 2 : 1
+        const dirname = `/${chunks.slice(2, -1 * (baseSliceEnd + (dependency['dependency-name'].match(/\//g) || []).length)).join('/') || ''}`
```

Comment:
The test cases LGTM and this PR makes sense, but I think we should refactor  dirname since `baseSliceEnd` adds some more complexity.

```suggestion
        // When a branch delimiter of "-" is used, we need to adjust for the hyphen
        // between the dependency name and version at the end of the branch name
        // e.g. "fsevents-1.2.13".
        const adjustForDelim = delim === '-' ? 2 : 1
        const numDirectorySegments =  (dependency['dependency-name'].match(/\//g) || []).length + adjustForDelim
        const dirname = `/${chunks.slice(2, -1 * numDirectorySegments).join('/') || ''}`
```

Reply:
I agree that calculating the directory name needs refactoring - will try to see if I can make it better

Reply:
Alright I have tried to refactor it into something that is hopefully easier to follow - open to feedback